test/e2e/e2e.bats

```diff
@@ -2,25 +2,82 @@
 
 source test/e2e/helpers.sh
 
+setup() {
+	load 'helpers/bats-support/load'
```

Comment:
As much as I like the syntax and the practical effects of the "bats-support" helper functions, I don't like the idea of copying the whole project. I think we could create our own helper functions and make use local tools like `grep`, `awk` and others.

Reply:
That seems a bit silly to basically re-write a library that is provided by the framework that we want to use.
I would be open to adding it as a submodule.

Reply:
hmmm ... I see both sides here

don't know the answer to this question:  even though bats is not a go project, can we use go mod to vendor the repo in, and then load from the vendored location?

Reply:
That seems a bit sketchy to me, I think I would rather the submodule.

Reply:
sorry multitasking too much this AM ... can you clarify / further explain what you mean by submodule ?

Reply:
using a git submodule to pull in those projects.

Reply:
ah *those* submodules :-)

yes that feels like the appropriate solution to achieve the same end goal as the go mod vendor idea ... just been forever since I even thought about git submodules

so yeah since you are open to the idea @coreydaley please do refactor this PR to pull in the bats-core helpers in that fashion, assuming using git submodules in our project have not blockers .... I have no idea if this is the case, but I would not be surprised if @adambkaplan as someone with more administrative powers to add the submodule to the "parent" project i.e. our cli repo here

Reply:
code has been updated to pull in BATS via git submodules and run without having to install BATS locally. ptal
Note that tests won't pass until the shipwright-io/build pr merges

Reply:
> hmmm ... I see both sides here
> 
> don't know the answer to this question: even though bats is not a go project, can we use go mod to vendor the repo in, and then load from the vendored location?

I like this idea! Would be easier to deal with than "git sub-modules" IMO, and it does solve the issue of keeping track of another external repository. 👍🏼

Reply:
looking at @coreydaley 's https://github.com/shipwright-io/cli/pull/27/commits/2150530e2c142c4bc2b7c9f791783324931a13f8 I actually agree with him @otaviof that we should use git submodules for this ... I agree go mod vendoring should not be used for a project like BATS which is not go based

Reply:
I think vendoring the external dependency is easier due being part of regular commits, we include the `vendor` directory in the project itself, as usual, and it makes everything simpler IMO. Git submodule is something I only have bad memories, and compared to the `vendor` option, it does require extra steps.

Reply:
Git sub-modules are really the de-facto way to include external libraries like this in your repository. 
Vendoring a project that has absolutely no Go code seems pretty sketchy.

Reply:
Then, I think we can benefit from a more in-depth discussion about the git-submodules, @coreydaley. Seems a bit too much given we need a couple of assertions we can easily achieve with shell-script.

Reply:
sounds like another topic for Monday / community meeting